python/paddle/fluid/transpiler/distribute_transpiler.py

```diff
@@ -547,8 +605,61 @@ def _get_splited_name_and_shape(varname):
                     inputs=new_inputs,
                     outputs=new_outputs,
                     attrs=op.attrs)
+                for var in new_outputs.values():
+                    load_vars.append(var.name)
+        # add checkpoint op
+        if not checkpoint_load_dir:
+            return s_prog
+
+        serial_number = self._get_lastest_checkpoint_dir(checkpoint_load_dir)
+
+        s_prog.global_block().append_op(
+            type="checkpoint_load",
+            inputs={"X": load_vars},
+            attrs={"dir": checkpoint_load_dir,
+                   "Serial": serial_number})
+
         return s_prog
 
+    def _is_persistable(self, var):
+        """only save LodTensor variable"""
+        if var.desc.type() == core.VarDesc.VarType.FEED_MINIBATCH or \
+                var.desc.type() == core.VarDesc.VarType.FETCH_LIST or \
+                var.desc.type() == core.VarDesc.VarType.RAW :
+            return False
+        return var.persistable
+
+    def _get_lastest_checkpoint_dir(self, checkpoint_dir):
+        """
+        get the biggest number in checkpoint_dir, which has _SUCCESS
+        """
+        if not checkpoint_dir.strip():
+            return ""
+
+        def has_success(checkpoint_dir, cur_dir):
+            """
+            is _SUCCESS in this dir
+            """
+            if not os.path.isdir(cur_dir):
+                return -1
+
+            try:
+                int(cur_dir)
+            except ValueError:
+                return -1
+
+            success_path = os.path.join(checkpoint_dir, cur_dir, SUCCESS)
```

Comment:
what is success_path used for?

Reply:
We need a tag to indicate that: the checkpoint content is right. So, I define a tag named ```_SUCCESS```, when the ```checkpoint_save_op``` save all need variables successfully, it will write an empty file named ```_SUCCESS``` at last.
Because of this, when pserver/trainer need to load ```checkpoint```, it will check ```_SUCCESS``` at first.

Reply:
If you only need to know when the saving ends, just wait for the `executor` returns, raise any error that may cause the saving fail is OK I think.

Reply:
If some exceptions happened in ```executor ``` running, the ```executor ``` does not have any information return, how do we know saving is OK.

Reply:
You can catch that exception I think, please give it a try, this will make the code more simple.